htdocs/expedition/class/expeditionbatch.class.php

```diff
@@ -207,6 +210,7 @@ static function fetchAll($db,$id_line_expdet)
 				$tmp->fk_origin_stock = $obj->fk_origin_stock;
 				$tmp->fk_expeditiondet = $obj->fk_expeditiondet;
 				$tmp->dluo_qty = $obj->qty;
+				$tmp->entrepot_id = $obj->fk_entrepot;
```

Comment:
This data come from a link that is ok just after the reception is done, but as soon as stock is empty, the line into product_batch and product_stock will no exists anymore and we won't be able to get the entrepot_id value for old reception. Is this ok for you ?

Reply:
I will revert this, thought I need the entrepot_id to easy get the warehouse source of the lot ready to ship, but I seem to have it resolved it in another way. I get it now with a fetch from product_batch. I need the warehouse from product_batch to get it's source location, you should not remove stock before validating. If you remove the lot from stock you also won't be able to validate the shipment. This is also an advantage of the new edit option. When you get the error the lot isn't available anymore, you can replace it by another lot for validating.